### PR TITLE
Added the capability to retrieve a specific version for a subject

### PR DESCRIPTION
**Pull request by jpardobl**
- Added method get_schema_by_version to confluent_kakfa.avro.CachedSchemaRegistryClient. This method retrieves a specific version for a subject.

 - Modified method get_latest_schema from confluent_kakfa.avro.CachedSchemaRegistryClient. Now this method uses method get_schema_by_version passing latest as the version to retrieve

  **** All the above changes keep total backward compatibility ***

### CODE REVIEW

**Comment by rnpridgeon** [comment]
Seems reasonable to me, please sign the CLA and extend the existing schema registry client integration tests to cover this new feature. 

https://github.com/confluentinc/confluent-kafka-python/blob/master/tests/integration/integration_test.py#L737

**Comment by jpardobl** [comment]
Already done both steps: added the tests and signed off the CLA. Thanks!

**Comment by cricket007** [comment]
Handled this in a bit safer way, that more closely matches the Java API. Ref #515

**Comment by rnpridgeon** [comment]
Thanks @cricket007 and @jpardobl. 

@cricket007 as discussed in pr #515 I don't think it's necessary to handle integer overflow issues on the client. Please feel free to follow up in the PR if you have further questions

@jpardobl in addition to the unit test you provided can you please add an integration test as well. You can simply call `get_compatibility` again with the `latest_id` returned earlier in the test and ensure you get the same value. 

https://github.com/confluentinc/confluent-kafka-python/blob/master/tests/integration/integration_test.py#L737-L756

File: confluent_kafka/avro/cached_schema_registry_client.py [hunk 1]
```diff
@@ -315,6 +316,25 @@ def get_latest_schema(self, subject):
 
         self._cache_schema(schema, schema_id, subject, version)
         return (schema_id, schema, version)
+    
+
+    def get_latest_schema(self, subject):
+        """
+        GET /subjects/(string: subject)/versions/(versionId: version)
```
Comment (cricket007) [inline confluent_kafka/avro/cached_schema_registry_client.py]: `(versionId: version)` should be 'latest' here

File: confluent_kafka/avro/cached_schema_registry_client.py [hunk 2]
```diff
@@ -291,7 +292,7 @@ def get_latest_schema(self, subject):
         :returns: (schema_id, schema, version)
         :rtype: (string, schema, int)
         """
-        url = '/'.join([self.url, 'subjects', subject, 'versions', 'latest'])
+        url = '/'.join([self.url, 'subjects', subject, 'versions', version])
```
Comment (cricket007) [inline confluent_kafka/avro/cached_schema_registry_client.py]: The join will fail if the version isn't a string, so best to put `str(version)`

File: confluent_kafka/avro/cached_schema_registry_client.py (hunk 1 above)
Comment (rooom13) [reply confluent_kafka/avro/cached_schema_registry_client.py]: why not adding a parameter "version" so we could retrieve the schema given a subject & the verssion?

**Comment by rooom13** [comment]
Tha issue would really help a lot

**Comment by rnpridgeon** [comment]
https://github.com/confluentinc/confluent-kafka-python/blob/master/confluent_kafka/schema_registry/schema_registry_client.py#L492

**Pull request closed without merging**